lib/pka_dev.c

```diff
@@ -1305,7 +1640,18 @@ static int pka_dev_init_shim(pka_dev_shim_t *shim)
 
     // Configure the TRNG
     ret = pka_dev_config_trng(&shim->resources.aic_csr,
-                                &shim->resources.trng_csr);
+                              &shim->resources.trng_csr);
+
+    // Run TRNG tests after configuring TRNG
+    if (!ret)
+        ret = pka_dev_test_trng(shim->resources.trng_csr.ioaddr,
+                                shim->resources.trng_csr.base);
+
+    // Re-Configure the TRNG after tests are successful for generic usage

```

Comment:
Why we need to re-configure the TRNG?

Reply:
I noticed that if we use TRNG after running the test circuits (without re-configuring), the timer waiting for 'output ready' gets done and we can never retrieve random output. I think this might be due to TRNG is left in an undefined state due to writing to testing related registers. Hence I thought TRNG needs to be re-configured.

Reply:
My understanding is that self-tests are intended to test the TRNG with a given configuration. If the test is successful then "continue running with the actual configuration". If we re-configure the TRNG then the self-tests results would be irrelevant as we might be running with a bogus configuration (even though, it is practically the same configuration). Does it make sense?

Reply:
Yes, you're right, that makes sense. During debugging I saw that if TRNG is enabled and then the tests are carried out, it tends to leave TRNG in a bad state and hence TRNG won't function. But, if configuration is done and TRNG is not enabled when carrying out the tests seems to be doing fine. Also, as per the INSIDE-Secure documentation, TRNG need not be enabled when carrying out these tests. Hence I have made the change to carry out the tests after configuring TRNG but before TRNG is enabled. Is this okay?